modules/diskq/diskq-global-metrics.c

```diff
@@ -63,6 +72,156 @@ _dir_watch_timer_start(DiskQGlobalMetrics *self)
   iv_timer_register(&self->dir_watch_timer);
 }
 
+static void
+_init_abandoned_disk_buffer_sc_keys(StatsClusterKey *queued_sc_key, StatsClusterKey *capacity_sc_key,
+                                    StatsClusterKey *disk_allocated_sc_key, StatsClusterKey *disk_usage_sc_key,
+                                    const gchar *abs_filename, gboolean reliable)
+{
+  enum { labels_len = 3 };
+  static StatsClusterLabel labels[labels_len];
+  labels[0] = stats_cluster_label("abandoned", "true");
+  labels[1] = stats_cluster_label("path", abs_filename);
+  labels[2] = stats_cluster_label("reliable", reliable ? "true" : "false");
+
+  stats_cluster_single_key_set(queued_sc_key, "disk_queue_events", labels, labels_len);
+
+  stats_cluster_single_key_set(capacity_sc_key, "disk_queue_capacity_bytes", labels, labels_len);
+  stats_cluster_single_key_add_unit(capacity_sc_key, SCU_KIB);
+
+  stats_cluster_single_key_set(disk_allocated_sc_key, "disk_queue_disk_allocated_bytes", labels, labels_len);
+  stats_cluster_single_key_add_unit(disk_allocated_sc_key, SCU_KIB);
+
+  stats_cluster_single_key_set(disk_usage_sc_key, "disk_queue_disk_usage_bytes", labels, labels_len);
+  stats_cluster_single_key_add_unit(disk_usage_sc_key, SCU_KIB);
+}
+
+static void
+_set_abandoned_disk_buffer_file_metrics(StatsClusterKey *queued_sc_key, StatsClusterKey *capacity_sc_key,
+                                        StatsClusterKey *disk_allocated_sc_key, StatsClusterKey *disk_usage_sc_key,
+                                        const gchar *dir, const gchar *abs_filename, gboolean reliable)
+{
+  DiskQueueOptions options;
+  disk_queue_options_set_default_options(&options);
+  disk_queue_options_set_dir(&options, dir);
+  options.reliable = reliable;
+  options.read_only = TRUE;
+
+  LogQueue *queue;
+  if (reliable)
+    queue = log_queue_disk_reliable_new(&options, abs_filename, NULL, STATS_LEVEL0, NULL, NULL);
+  else
+    queue = log_queue_disk_non_reliable_new(&options, abs_filename, NULL, STATS_LEVEL0, NULL, NULL);
+  LogQueueDisk *queue_disk = (LogQueueDisk *) queue;
+
+  if (!log_queue_disk_start(queue))
```

Comment:
This has some `internal()` side effects on the info level:
```
[2023-04-05T11:57:37.812272] Disk-buffer state loaded; filename='/tmp/asd/syslog-ng-00abandoned000.qf', number_of_messages='0'
```

I don't think it's a problem. Please just click "resolve" if you agree.